functions/ts/src/istioctl_analyze.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Configs,
+  FunctionConfigError,
+  configFileResult,
+  kubernetesObjectResult,
+  Severity,
+} from 'kpt-functions';
+import { spawnSync } from 'child_process';
+import { isConfigMap } from './gen/io.k8s.api.core.v1';
+
+const FLAG_ARGS = 'flags';
+const USE_KUBE_FLAG = '--use-kube';
+const OUTPUT_SHORT_FLAG = '-o';
+const OUTPUT_LONG_FLAG = '--output';
+
+interface IstioResult {
+  code: string;
+  level: 'Error' | 'Warn' | 'Info';
+  origin: string;
+  reference: string;
+  message: string;
+  documentation_url: string;
+}
+
+// Analyze istio configs using istioctl analyze.
+export async function istioctlAnalyze(configs: Configs) {
+  // Validate config data and read arguments.
+  const args = readArguments(configs);
+
+  for (const object of configs.getAll()) {
+    try {
+      const child = spawnSync('istioctl', args, {
+        input: JSON.stringify(object),
+        encoding: 'utf-8',
+      });
+      const error = child.stderr;
+      if (error && error.length > 0) {
+        configs.addResults(
+          configFileResult(
+            `Istioctl analyze command results in error: ${error}`,
+            '',
+            'error'
+          )
+        );
+      }
+      if (child.stdout && child.stdout !== 'null') {
+        const output: IstioResult[] = JSON.parse(child.stdout);
+        if (output && output.length) {
+          output.forEach(istioResult => {
+            const result = kubernetesObjectResult(
+              istioResult.message,
+              object,
+              undefined,
+              istioResult.level.toLowerCase() as Severity
+            );
+            result.tags = {
+              ['documentation_url']: istioResult.documentation_url,
+              ['origin']: istioResult.origin,
+              ['code']: istioResult.code,
+            };
+            configs.addResults(result);
+          });
+        }
+      }
+    } catch (err) {
+      configs.addResults(configFileResult(`${err}`, '', 'error'));
+    }
+  }
+}
+
+function readArguments(configs: Configs) {
+  // Initialize to output json
+  const args: string[] = ['analyze', '-', '-o', 'json'];
+  const data = readConfigDataOrThrow(configs);
+  for (const key in data) {
+    if (key === FLAG_ARGS) {
+      args.push(data[key]);
+    } else if (key === OUTPUT_SHORT_FLAG || key === OUTPUT_LONG_FLAG) {
+      continue;
```

Comment:
?

if you're ignoring these, then you don't the else if

Reply:
I do need this otherwise it will fall into the data.hasOwnProperty condition.